go/mcap/range_index_heap.go

```diff
@@ -0,0 +1,54 @@
+package mcap
+
+// rangeIndex contains either a ChunkIndex or a MessageIndexEntry to be sorted on LogTime.
```

Comment:
what's the rationale for this structure containing either chunk indexes or message indexes?

Reply:
To order them against eachother, and decide per-message if we need to read a message out of an existing chunk or decompress a new one.